deploy/terraform/main.tf

```diff
@@ -0,0 +1,101 @@
+provider "kubernetes" {
+  config_path = "~/.kube/config"
+}
+
+resource "kubernetes_namespace" "mayastor_ns" {
+  metadata {
+    name = "mayastor"
+  }
+}
+
+resource "kubernetes_secret" "regcred" {
+  metadata {
+    name      = "regcred"
+    namespace = "mayastor"
+  }
+  data = {
+    ".dockerconfigjson" = "${file("~/.docker/config.json")}"
+  }
+
+  type = "kubernetes.io/dockerconfigjson"
+  depends_on = [
+    kubernetes_namespace.mayastor_ns,
+  ]
+}
+
+
+module "nats" {
+  source     = "./mod/nats"
+  nats_image = var.nats_image
+  depends_on = [
+    kubernetes_namespace.mayastor_ns,
+    kubernetes_secret.regcred
+  ]
+}
+
+module "rbac" {
+  source = "./mod/rbac"
+}
+
+module "etcd" {
+  source = "./mod/etcd"
+  image  = var.etcd_image
+  depends_on = [
+    kubernetes_namespace.mayastor_ns,
+    kubernetes_secret.regcred
+  ]
+}
+
+module "csi-agent" {
+  source         = "./mod/csi-agent"
+  agent_image    = var.csi_agent_image
+  registar_image = var.csi_registar_image
+}
+
+module "msp-operator" {
+  source = "./mod/k8s-operator"
+  image  = var.msp_operator_image
+  depends_on = [
+    module.rbac,
+    module.core,
+    module.rest
+  ]
+}
+
+module "rest" {
```

Comment:
Should rest depend on `module.nats` and `module.core` as that is what it is communicating with?

Reply:
Yes, but good news; it works without it :) I did that purposely to test the dependency stuff. But, in principle: yes it should wait.